### PR TITLE
Add Harbor GituHub Actions

### PR DESCRIPTION
**Pull request by s225904**
# Description
Change build.yml file in workflow directory, adding github action for Harbor. Update also the DockerHub docker/build-push-action@v2 part because the issue 252 has been update

# How Has This Been Tested?
I've run this workflow many times and I've seen that the github actions when successfully ends push the images in the correct project. Unfortunately there is a problem with the robot account and some times a random github actions fail.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -108,71 +108,90 @@ jobs:
             context: ./operators
             dockerfile: ./operators/build/golang-common/Dockerfile
             build-args: COMPONENT=instance-operator
+            harbor-args: crownlabs-core
```
Comment (giorio94) [inline .github/workflows/build.yml:111]:
```suggestion
            harbor-project: crownlabs-core
```

Same below

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -204,10 +223,19 @@ jobs:
...
+          registry: https://harbor.crownlabs.polito.it
+          username: ${{ secrets.HARBOR_REGISTRY_USERNAME }}
+          password: ${{ secrets.HARBOR_REGISTRY_PASSWORD }}
+        if: needs.configure.outputs.repo-push == 'true'
+
       - name: Configure the build-push-action parameters
         id: parameters
         env:
           DOCKER_USERNAME: ${{ secrets.DOCKER_USERNAME }}
+          HARBOR_USERNAME: ${{ secrets.HARBOR_REGISTRY_USERNAME }}
```
Comment (giorio94) [inline .github/workflows/build.yml:238]: This is not required

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -204,10 +223,19 @@ jobs:
           password: ${{ secrets.DOCKER_PASSWORD }}
         if: needs.configure.outputs.repo-push == 'true'
 
+      - name: Login to Harbor
+        uses: docker/login-action@v1
+        with:
+          registry: https://harbor.crownlabs.polito.it
```
Comment (giorio94) [inline .github/workflows/build.yml:229]: I created the secret `HARBOR_REGISTRY_URL` to avoid having to hard-code this.

**Review by giorio94 — COMMENTED** [review]
A couple of comments inline

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -218,25 +245,22 @@ jobs:
 
       - name: Build and Push the ${{ matrix.component }} image
         uses: docker/build-push-action@v2
+        env:
+          HARBOR_URL: harbor.crownlabs.polito.it
```
Comment (giorio94) [inline .github/workflows/build.yml:249]: Should use the secret, instead of the hardcoded value

**Review by giorio94 — APPROVED** [review]
Just a minor comment

**Review by s225904 — COMMENTED** [review]
fixed

**Pull request merged**